packages/amplify-category-api/src/provider-utils/awscloudformation/cfn-api-artifact-handler.ts

```diff
@@ -129,6 +129,30 @@ class CfnApiArtifactHandler implements ApiArtifactHandler {
     printApiKeyWarnings(this.context, oldConfigHadApiKey, authConfigHasApiKey(authConfig));
   };
 
+  updateArtifactsWithoutCompile = async (request: UpdateApiRequest): Promise<void> => {
```

Comment:
is there a reason we would not want to compile?

Reply:
Originally it was so that the compiler didn't run twice on amplify push when updating the API Key, but I think we can consolidate this work with the fix I am doing on the transformer branch. I'll update this.

Reply:
@SwaySway I removed the `updateArtifactsWithoutCompile` and am just using `updateArtifacts` for now.

Reply:
> Originally it was so that the compiler didn't run twice on amplify push when updating the API Key, but I think we can consolidate this work with the fix I am doing on the transformer branch. I'll update this.

Ah that makes sense, maybe we can add a flag into the update artifacts to not run gql-compile. I think there might be some value in add that, though I'd imagine we'd need to review this.